lib/maths/common/Makefile

```diff
@@ -0,0 +1,83 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License
+# 2.0 and the following additional limitation. Functionality enabled by the
+# files subject to the Elastic License 2.0 may only be used in production when
+# invoked by an Elasticsearch process with a license key installed that permits
+# use of machine learning features. You may not use this file except in
+# compliance with the Elastic License 2.0 and the foregoing additional
+# limitation.
+#
+include $(CPP_SRC_HOME)/mk/defines.mk
+
+TARGET=$(OBJS_DIR)/libMlMathsCommon$(DYNAMIC_LIB_EXT)
+
+USE_BOOST=1
+USE_BOOST_FILESYSTEM_LIBS=1
```

Comment:
Are you sure this is needed for all 3 split libraries?  It seems like it might not be widely used within the maths library.